bundles/config/org.eclipse.smarthome.config.dispatch/src/main/java/org/eclipse/smarthome/config/dispatch/internal/ConfigDispatcher.java

```diff
@@ -157,8 +158,9 @@ protected boolean watchSubDirectories() {
      * (java.nio.file.Path)
      */
     @Override
-    protected void registerDirectory(Path subDir) throws IOException {
-        subDir.register(watchService, ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY);
+    protected WatchKey registerDirectory(Path subDir) throws IOException {
+        WatchKey registrationKey = subDir.register(watchService, ENTRY_CREATE, ENTRY_DELETE, ENTRY_MODIFY);

```

Comment:
What do you think about:
* http://pmd.sourceforge.net/pmd-4.3.0/rules/design.html#UnnecessaryLocalBeforeReturn
* https://dev.eclipse.org/sonar/coding_rules#rule_key=squid%3AS1488|s=createdAt|asc=false